src/core/border_router/routing_manager.cpp

```diff
@@ -406,98 +406,76 @@ void RoutingManager::UpdateDiscoveredPrefixTableOnNetDataChange(void)
     mDiscoveredPrefixTable.SetAllowDefaultRouteInNetData(foundDefRouteOmrPrefix);
 }
 
-void RoutingManager::EvaluateOmrPrefix(OmrPrefixArray &aNewOmrPrefixes)
+void RoutingManager::EvaluateOmrPrefix(OnMeshPrefixArray &aNewPrefixes)
```

Comment:
nit:
```suggestion
void RoutingManager::EvaluateOnMeshPrefix(OnMeshPrefixArray &aNewPrefixes)
```

Reply:
I was considering a rename here but decided to keep the name as `EvaluateOmrPrefix()` for couple of reasons:
- One aspect of this method is to determine the currently favored OMR prefix (from net data) and decide whether we need to add/remove our local OMR prefix to net data (so we are evaluating OMR prefix in this method)
- The other aspect is to populate the `OnMehsPrefixArray` which we later include as RIO when we sent RA.
- I have a follow-up PR/change where we separate these two aspects and have `SendRouterAdvertisement()` directly determine the list  of prefixes to advertise (we use priority for which prefixes to include first as discussed in https://github.com/openthread/openthread/pull/7900#issuecomment-1187720201 and [SPEC-1087](https://threadgroup.atlassian.net/browse/SPEC-1087))
- I think this will help make the code simpler.

Reply:
Sounds good, thanks!